src/main/java/unicash/model/budget/BudgetList.java

```diff
@@ -0,0 +1,62 @@
+package unicash.model.budget;
+
+import static java.util.Objects.requireNonNull;
+import static unicash.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Iterator;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import unicash.model.budget.exceptions.BudgetNotFoundException;
+
+
+/**
+ * A list of budgets that enforces uniqueness between its elements and does not allow nulls.
+ */
+public class BudgetList implements Iterable<Budget> {
```

Comment:
Not sure why we need a BudgetList if we're enforcing a single budget across the application :o

Reply:
Hmm, okay, sure, I think we might not need to focus on expansion in the long term since we're not really adding more features after v1.3 but we can work with 3 different types of budget. In that case, do you just want to hardcode these values? Like for a daily, weekly and monthly budget?

Reply:
@jamz903 I think it's good to extend budget to specific categories, because one of the criteria for product design is specificity to university students, and with a category-specific budget we could claim that students can set specific budget according to their lifestyles, like one for SchoolExpenses (#school), CCAExpenses (#cca), one for Social Events (#social), of course non-target-users can use this feature as well, but with specific budgets it would be a better fit for our target users specifically! 

Our target users would most definitely also benefit from Daily, Weekly, Monthly budget, and we can claim that these standard budgets are unique to our target users only, but additionally having category-specific budgetting would be a better fit I feel!

Reply:
@woojiahao I think we should stick to BudgetList, instead of hardcoding the values. Would be easier to build on a budgetList rather than refactor the code later on should we decide to implement budget for categories!

Reply:
Hm, alright, makes sense, will figure out a way to weave this into the storage too as well then

Reply:
Update after weekly meeting on 27 Oct: Decided on a singular universal budget for easier implementation